igb/train_multi_hetero.py

```diff
@@ -0,0 +1,231 @@
+                )
+
+        sched.step()
+
+    model.eval()
+    if proc_id == 0:
+        predictions = []
+        labels = []
+        with torch.no_grad():
+            for _, _, blocks in test_dataloader:
+                blocks = [block.to(device) for block in blocks]
+                inputs = blocks[0].srcdata['feat']
+                labels.append(blocks[-1].dstdata['label'][category].cpu().numpy())
+                predictions.append(model(blocks, inputs).argmax(1).cpu().numpy())
+            predictions = np.concatenate(predictions)
+            labels = np.concatenate(labels)
+            test_acc = sklearn.metrics.accuracy_score(labels, predictions)*100
+        print("Test Acc {:.2f}%".format(test_acc))
+        print("Total time taken " + str(datetime.timedelta(seconds = int(time.time() - training_start))))
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    # Loading dataset
+    parser.add_argument('--path', type=str, default='/mnt/nvme14/IGB260M/', 
+        help='path containing the datasets')
+    parser.add_argument('--dataset_size', type=str, default='tiny',
+        choices=['tiny', 'small', 'medium', 'large', 'full'], 
+        help='size of the datasets')
+    parser.add_argument('--num_classes', type=int, default=19, 
+        choices=[19, 2983], help='number of classes')
+    parser.add_argument('--in_memory', type=int, default=0, 
+        choices=[0, 1], help='0:read only mmap_mode=r, 1:load into memory')
+    parser.add_argument('--synthetic', type=int, default=0,
+        choices=[0, 1], help='0:nlp-node embeddings, 1:random')
+
+    # Model
+    parser.add_argument('--model_type', type=str, default='rgat',
+                        choices=['rgat', 'rsage', 'rgcn'])
+    parser.add_argument('--modelpath', type=str, default='deletethis.pt')
+    parser.add_argument('--model_save', type=int, default=0)
+
+    # Model parameters 
+    parser.add_argument('--fan_out', type=str, default='10,15')
+    parser.add_argument('--batch_size', type=int, default=10240)
+    parser.add_argument('--num_workers', type=int, default=4)
+    parser.add_argument('--hidden_channels', type=int, default=128)
+    parser.add_argument('--learning_rate', type=int, default=0.01)
+    parser.add_argument('--decay', type=int, default=0.001)
+    parser.add_argument('--epochs', type=int, default=20)
+    parser.add_argument('--num_layers', type=int, default=6)
+    parser.add_argument('--num_heads', type=int, default=4)
+
+    parser.add_argument("--sched_stepsize", type=int, default=25)
+    parser.add_argument("--sched_gamma", type=float, default=0.25)
+
+    parser.add_argument('--log_every', type=int, default=5)
+    parser.add_argument('--device', type=int, default=0)
+
+    parser.add_argument('--gpu_devices', type=str, default='0,1,2,3')
+    args = parser.parse_args()
+
+    gpu_idx = [int(fanout) for fanout in args.gpu_devices.split(',')]
+    num_gpus = len(gpu_idx)
+    dataset = IGBHeteroDGLDataset(args)
```

Comment:
At this line, we are utilizing the `IGBHeteroDGLDataset` instead of `IGBHeteroDGLDatasetMassive`, same as the implementation in [`train_hetero.py`](https://github.com/IllinoisGraphBenchmark/IGB-Datasets/blob/28ca6b343d75c170e515e62d5e4f9f03d5553f77/igb/train_hetero.py#L182), so this script is not verified for large and full dataset.